_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -236,80 +238,125 @@ The only element with a header [role][semantic role] is part of a table which is
 
 #### Inapplicable Example 2
 
-The only element with header [role][semantic role] is part of a table which is not [visible][].
+There are no element with a [semantic role][] of header.
 
 ```html
-<html>
-	<style>
-		.notInPage {
-			position: absolute;
-			left: -9999px;
-			top: -9999px;
-		}
-	</style>
-	<body>
-		<table class="notInPage">
-			<tr>
-				<th>Time</th>
-			</tr>
-			<tr>
-				<td>24:00</td>
-			</tr>
-		</table>
-	</body>
-</html>
+<table></table>
 ```
 
 #### Inapplicable Example 3
 
-There is no element with a [semantic role][] of header within the table.
+There are no elements with a [role][semantic role] of header within the table.
 
 ```html
 <table>
 	<tr>
-		<td>12:00</td>
+		<th role="cell">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
 	</tr>
 </table>
 ```
 
 #### Inapplicable Example 4
 
-There are no [descendants][descendant] within the table.
+The only element with a [role][semantic role] of header is not [visible][].
 
 ```html
-<table></table>
+<table>
+	<tr>
+		<th style="display: none;">Column A</th>
+	</tr>
+	<tr>
+		<td>Cell A</td>
+	</tr>
+</table>
 ```
 
 #### Inapplicable Example 5
 
-The only element with a [role][semantic role] of header is not [visible][].
+The only element with a [role][semantic role] of header is not [included in the accessibility tree][].
 
 ```html
 <table>
 	<tr>
-		<th style="display: none;"></th>
+		<th aria-hidden="true">Column A</th>
 	</tr>
 	<tr>
-		<td></td>
+		<td>Cell A</td>
 	</tr>
 </table>
 ```
 
 #### Inapplicable Example 6
 
-There are no elements with a [role][semantic role] of header within the table.
+The only element with a [role][semantic role] of header does not have a closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]
+
+```html
+<div>
+	<tr>
+		<th>Column A</th>
+	</tr>
+</div>
+```
+
+#### Inapplicable Example 7
+
+The only element with header [role][semantic role] is part of a table which has [descendants][descendant] that are empty ("").
```

Comment:
This is inapplicable because the `header` is not visible (so, same reason as inapplicable example 4).
We need an inapplicable example where the only reason for which it's inapplicable is that the table has no non-empty descendants.